commit bbc806242f6d334cd5fc60b45c64eddb9310d6ed (refs/changes/34/334/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-08-22T18:19:36-07:00 (3 years, 2 months ago)
    
    AGENT-1029 CN agent reports negative disk_system_used_bytes values in sysinfo

diff --git a/lib/heartbeater.js b/lib/heartbeater.js
index 24f303a..e09dd25 100644
--- a/lib/heartbeater.js
+++ b/lib/heartbeater.js
@@ -518,17 +518,11 @@ StatusReporter.prototype.gatherDiskUsage = function (vms, callback) {
         },
         function (cb) {
             // #7
-            zpool.list(
-                'zones', { parseable: true, fields: ['size', 'alloc'] },
-                function (error, fields, pools) {
-                    if (error) {
-                        cb(error);
-                        return;
-                    }
-                    usage.pool_size_bytes = toInt(pools[0][0]);
-                    usage.pool_alloc_bytes = toInt(pools[0][1]);
-                    cb();
-                });
+            var poolds = datasets['zones'];
+            usage.pool_alloc_bytes = toInt(poolds.used);
+            usage.pool_size_bytes = toInt(poolds.used) +
+                toInt(poolds.available);
+            cb();
         },
         function (cb) {
             // #8
